tests/cpp/unit-tests/CMakeLists.txt

```diff
@@ -31,110 +31,27 @@ cmake_minimum_required(VERSION 3.13)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_BUILD_TYPE DEBUG)
 set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_FLAGS --coverage)
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_C_COMPILER_ID STREQUAL "GNU"))
+    set(CMAKE_CXX_FLAGS --coverage)
+    add_link_options(--coverage)
+endif()
```

Comment:
See previous comment

Reply:
Now this is contingent on -DCOVERAGE=on being passed to the cmake command